From d30418d29bc7b171d6612df7e2bb9f914e954d41 Mon Sep 17 00:00:00 2001
From: Dan Williams <dan.j.williams@intel.com>
Date: Tue, 3 Mar 2020 14:58:35 -0800
Subject: [PATCH] ndctl/test: Relax dax_pmem_compat requirement

Patch-mainline: v68
Git-commit: b7991dbc22f31d03a38f3ee2dca4446fb55279e3

While there are some tests that require the new "dax-bus" device model,
none of the tests require compatibility mode. Drop the requirement so
the tests work with DEV_DAX_PMEM_COMPAT=n kernels.

Reviewed-by: Jan Kara <jack@suse.cz>
Cc: Jan Kara <jack@suse.cz>
Link: http://lore.kernel.org/r/20200123154720.12097-1-jack@suse.cz
Link: https://lore.kernel.org/r/158327631566.2222444.16879386597302511191.stgit@dwillia2-desk3.amr.corp.intel.com
Signed-off-by: Dan Williams <dan.j.williams@intel.com>
---
 test/core.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/test/core.c b/test/core.c
index 4b36b2d14d7b..8aae685afc2a 100644
--- a/test/core.c
+++ b/test/core.c
@@ -168,6 +168,14 @@ int nfit_test_init(struct kmod_ctx **ctx, struct kmod_module **mod,
 			continue;
 retry:
 		rc = kmod_module_new_from_name(*ctx, name, mod);
+
+		/*
+		 * dax_pmem_compat is not required, missing is ok,
+		 * present-but-production is not ok.
+		 */
+		if (rc && strcmp(name, "dax_pmem_compat") == 0)
+			continue;
+
 		if (rc) {
 			log_err(&log_ctx, "%s.ko: missing\n", name);
 			break;
-- 
2.23.0

